rpcs3/Emu/Cell/Modules/cellGame.cpp

```diff
@@ -786,14 +786,78 @@ error_code cellGameCreateGameData(vm::ptr<CellGameSetInitParams> init, vm::ptr<c
 
 error_code cellGameDeleteGameData(vm::cptr<char> dirName)
 {
-	cellGame.todo("cellGameDeleteGameData(dirName=%s)", dirName);
+	cellGame.warning("cellGameDeleteGameData(dirName=%s)", dirName);
 
 	if (!dirName)
 	{
 		return CELL_GAME_ERROR_PARAM;
 	}
 
-	return CELL_OK;
+	const std::string name = dirName.get_ptr();
+	const std::string dir = vfs::get("/dev_hdd0/game/"s + name);
+
+	const auto prm = g_fxo->get<content_permission>();
+
+	auto remove_gd = [&]() -> error_code
+	{
+		if (Emu.GetCat() == "GD" && Emu.GetTitleID() == name)
+		{
+			// Cannot delete its own directory
+			return CELL_GAME_ERROR_NOTSUPPORTED;
+		}
+
+		psf::registry sfo = psf::load_object(fs::file(dir + "/PARAM.SFO"));
+
+		if (psf::get_string(sfo, "CATEGORY") != "GD" && fs::is_file(dir + "/PARAM.SFO"))
+		{
+			return CELL_GAMEDATA_ERROR_BROKEN;
+		}
+
+		if (sfo.empty())
+		{
+			// Nothing to remove
+			return CELL_GAME_ERROR_NOTFOUND;
+		}
+
+		// Actually remove game data
+		if (!vfs::host::remove_all(dir, Emu.GetHddDir(), true))
+		{
+			return {CELL_GAME_ERROR_ACCESS_ERROR, dir};
+		}
+
+		return CELL_OK;
+	};
+
+	while (true)
+	{
+		// Obtain exclusive lock and cancel init
+		auto _init = prm->init.init();
+
+		if (!_init)
+		{
```

Comment:
Is this guaranteed to be be false in the consecutive tries?

Reply:
no, why?

Reply:
Infact the whole point of the loop is to recheck in case something changed.